File: yanshaniOS/yanshaniOS/YanShanUI/Controller/BaseViewController/YSBaseViewController.m
//
//  YSBaseViewController.m
//  yanshaniOS
//
//  Created by 代健 on 2017/10/4.
//  Copyright © 2017年 jiandai. All rights reserved.
//

#import "YSBaseViewController.h"

@interface YSBaseViewController ()

@end

@implementation YSBaseViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor whiteColor];
    // Do any additional setup after loading the view.
    [self configViewControllerParameter];
    [self configContainer];
    [self configView];
    [self addNavigationItems];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)dealloc {
    
    NSLog(@"%@ release",[self class]);
    [[NSNotificationCenter defaultCenter] removeObserver:self];
}

#pragma mark - config view controller

- (void)configViewControllerParameter {
    // 子类重写实现
}

- (void)configView {
    // 子类重写实现
}

- (void)configContainer {
    self.edgesForExtendedLayout = UIRectEdgeNone;
    self.extendedLayoutIncludesOpaqueBars = NO;
    self.navigationController.navigationBar.translucent = NO;
    self.tabBarController.tabBar.translucent = NO;
    self.navigationController.interactivePopGestureRecognizer.delegate = (id)self;
}

- (void)addNavigationItems {
    
}

- (void)backViewController:(UIButton *)sender {
    [self.navigationController popViewControllerAnimated:YES];
}

@end

File: YanShanAnJianSwiftVersion/YanShanAnJianSwiftVersion/BaseClass/BaseViewController/YSBaseViewController.swift
//
//  YSBaseViewController.swift
//  YanShanAnJianSwiftVersion
//
//  Created by 代健 on 2018/6/19.
//  Copyright © 2018年 jiandai. All rights reserved.
//

import UIKit

class YSBaseViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func configViewControllerParameter() {
        //子类实现
    }
    
    func configViewLayout() {
        self.extendedLayoutIncludesOpaqueBars = false
        self.navigationController?.navigationBar.isTranslucent = false
    }
    
    func addSubViews() {
        
    }
    
    func addNavigationItems() {
        
    }
    
    func backToBeforeViewController() {
        self.navigationController?.popViewController(animated: true)
    }
    
    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
